Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp

```diff
@@ -1429,8 +1429,14 @@ void InstrumentDefinitionParser::createStructuredDetector(
   Element *pElem = nullptr;
   NodeIterator tags(pCompElem->ownerDocument(), NodeFilter::SHOW_ELEMENT);
   Node *pNode = tags.nextNode();
+  std::string beamDir = "";
 
   while (pNode) {
```

Comment:
You are re-reading information that has already been extracted. The `ReferenceFrame` is parsed and attached to the instrument before any detectors are load. See [here](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp#L965) called from [here](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp#L213). Use the in-memory reference frame not duplicated parsing functionality.